assets/js/modules/adsense/components/dashboard/AdSenseDashboardWidget.js

```diff
@@ -35,39 +35,44 @@ import getNoDataComponent from '../../../../components/legacy-notifications/noda
 import getDataErrorComponent from '../../../../components/legacy-notifications/data-error';
 import ProgressBar from '../../../../components/ProgressBar';
 import ModuleSettingsWarning from '../../../../components/legacy-notifications/module-settings-warning';
-import { getModulesData } from '../../../../util';
 import DateRangeSelector from '../../../../components/DateRangeSelector';
 import HelpLink from '../../../../components/HelpLink';
 import Header from '../../../../components/Header';
 import PageHeader from '../../../../components/PageHeader';
 import Layout from '../../../../components/layout/Layout';
 import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import { CORE_MODULES } from '../../../../googlesitekit/modules/datastore/constants';
+import { MODULES_ADSENSE, DATE_RANGE_OFFSET } from '../../../adsense/datastore/constants';
 import { Cell, Grid, Row } from '../../../../material-components';
 import { getCurrentDateRangeDayCount } from '../../../../util/date-range';
+import { generateDateRangeArgs } from '../../../adsense/util/report-date-range-args';
 
-const { withSelect } = Data;
+const { useSelect } = Data;
 
 // Empty component to allow filtering in refactored version.
 const AdSenseDashboardZeroData = withFilters( 'googlesitekit.AdSenseDashboardZeroData' )( () => null );
 
-class AdSenseDashboardWidget extends Component {
-	constructor( props ) {
-		super( props );
-		this.state = {
-			receivingData: true,
-			error: false,
-			zeroData: false,
-			loading: true,
-		};
-		this.handleDataError = this.handleDataError.bind( this );
-		this.handleDataSuccess = this.handleDataSuccess.bind( this );
-		this.handleZeroData = this.handleZeroData.bind( this );
-	}
+export default function AdSenseDashboardWidget() {
+	const [ receivingData, setReceivingData ] = useState( true );
+	const [ error, setError ] = useState( false );
+	const [ errorObj, setErrorObj ] = useState();
+	const [ loading, setLoading ] = useState( true );
+	const [ zeroData, setZeroData ] = useState( false );
+
+	const adSenseModuleConnected = useSelect( ( select ) => select( CORE_MODULES ).isModuleConnected( 'adsense' ) );
+	const dateRangeDates = useSelect( ( select ) => select( CORE_USER ).getDateRangeDates( {
+		offsetDays: DATE_RANGE_OFFSET,
+	} ) );
+	const accountSiteURL = useSelect( ( select ) => select( MODULES_ADSENSE ).getServiceAccountSiteURL(
+		{ ...generateDateRangeArgs( dateRangeDates ) }
+	) );
+	const dateRange = useSelect( ( select ) => select( CORE_USER ).getDateRange() );
+	const currentDayCount = getCurrentDateRangeDayCount( dateRange );
 
 	/**
-	 * Handle data errors from the contained AdSense component(s).
+	 * Handles data errors from the contained Analytics component(s).
```

Comment:
```suggestion
	 * Handles data errors from the contained AdSense component(s).
```

Reply:
Great spot. I was refactoring this based on a similar file and these comment changes must have crept in from there.